### PR TITLE
DatePicker: Adding callout props, and updating DatePicker Examples

### PR DESCRIPTION
**Pull request by kellygorr**
#### Pull request checklist

 [0] Addresses an existing issue: Fixes #0000
[X] Include a change request file using `$ npm run change`

#### Description of changes
Updating DatePicker to allow for calloutProps.  Fixing DatePicker examples and deprecating extra props.

This is a continuation of PR: https://github.com/OfficeDev/office-ui-fabric-react/pull/3465

**It looks like we don't have a need for calloutprops.  Removing PR**

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -194,10 +194,18 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
       className,
       minDate,
       maxDate,
-      calendarProps
+      calendarProps,
+      calloutProps
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx]: This is exposing/leaking the fact that the DatePicker uses a callout to consumers. Which props do you need to be able to set in the callout? It seems better to just expose the specific props in the component itself.

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -194,10 +194,18 @@ export class DatePicker extends BaseComponent<IDatePickerProps, IDatePickerState
...
     } = this.props;
     const { isDatePickerShown, formattedDate, selectedDate, errorMessage } = this.state;
 
+    let calendarStrings = null;
+    if (calendarProps && calendarProps.strings) {
+      calendarStrings = calendarProps.strings;
+    } else if (DEFAULT_STRINGS) {
+      calendarStrings = DEFAULT_STRINGS;
+    }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx]: The old code seems cleaner, why do you need to change?

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts [hunk 3]
```diff
@@ -20,6 +21,11 @@ export interface IDatePickerProps extends React.Props<DatePicker> {
    */
   calendarProps?: ICalendarProps;
 
+  /**
+  * Pass callout props to callout component
+  */
+  calloutProps?: ICallout;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: See my other comment, you are leaking the fact that the picker uses ICallout... If you want to set just a few props, may be best to just expose the props that you need to allow in date picker.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -58,7 +58,7 @@ Callout/  @joschect
 # ComboBox/
 CommandBar/   @micahgodbolt
 ContextualMenu/  @joschect
-DatePicker/   @johannao76 lorejoh12@gmail.com jolore@microsoft.com
+DatePicker/   @johannao76 lorejoh12@gmail.com jolore@microsoft.com kegorr@microsoft.com
```
Comment (johannao76) [inline .github/CODEOWNERS]: Why this change?

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts [hunk 5]
```diff
@@ -173,66 +187,75 @@ export interface IDatePickerProps extends React.Props<DatePicker> {
...
 
   /**
-   * An array of strings for the initials of the days of the week.
-   * The array is 0-based, so days[0] should be the initial of Sunday.
+   * An array of strings for the full names of months.
+   * The array is 0-based, so months[0] should be the full name of January.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
-  shortDays: string[];
+  months?: string[];
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: Instead of deprecating, it seems better to just make IDatePickerStrings extend ICalendarStrings, this way you don't have to change the existing API. You should make sure that the interface to existing consumer will not break also.

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts [hunk 6]
```diff
@@ -173,66 +187,75 @@ export interface IDatePickerProps extends React.Props<DatePicker> {
...
 
   /**
    * Aria-label for the "previous year" button.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
    */
   prevYearAriaLabel?: string;
 
   /**
    * Aria-label for the "next year" button.
+   * @deprecated Use 'calendarProps' with 'strings: ICalendarStrings' instead
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.types.ts]: See my other comment. You should try to rarely ever use deprecated, as changing the existing interface is generally not good.

File: packages/office-ui-fabric-react/src/components/DatePicker/examples/DatePicker.Required.Example.tsx [hunk 7]
```diff
@@ -81,12 +84,20 @@ export class DatePickerRequiredExample extends React.Component<{}, IDatePickerRe
   }
 
   public render() {
+<<<<<<< HEAD
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/DatePicker/examples/DatePicker.Required.Example.tsx]: Merge conflicts?

File: .github/CODEOWNERS (hunk 4 above)
Comment (kellygorr) [reply .github/CODEOWNERS]: I thought it would help to have a third person, I'll remove it!

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx (hunk 1 above)
Comment (kellygorr) [reply packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx]: I don't remember.  I will remove it.

File: packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx (hunk 2 above)
Comment (kellygorr) [reply packages/office-ui-fabric-react/src/components/DatePicker/DatePicker.tsx]: If someone uses calendarProps.strings then Default_strings overrides it.  I suppose this is ok, since Default strings is practically the same thing.

**Comment by kellygorr** [comment]
It looks like we don't have a need for calloutprops.  If this is the case and there is no need to deprecate anything then Removing PR!
